internal/kibana/connector_data_source.go

```diff
@@ -0,0 +1,90 @@
+package kibana
+
+import (
+	"context"
+
+	"github.com/elastic/terraform-provider-elasticstack/internal/clients"
+	"github.com/elastic/terraform-provider-elasticstack/internal/clients/kibana"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func DataSourceConnector() *schema.Resource {
+	var connectorSchema = map[string]*schema.Schema{
+		"connector_id": {
+			Description: "A UUID v1 or v4 randomly generated ID.",
+			Type:        schema.TypeString,
+			Computed:    true,
+		},
+		"space_id": {
+			Description: "An identifier for the space. If space_id is not provided, the default space is used.",
+			Type:        schema.TypeString,
+			Optional:    true,
+			Default:     "default",
+		},
+		"name": {
+			Description: "The name of the connector. While this name does not have to be unique, a distinctive name can help you identify a connector.",
+			Type:        schema.TypeString,
+			Required:    true,
+		},
+		"connector_type_id": {
+			Description: "The ID of the connector type, e.g. `.index`.",
+			Type:        schema.TypeString,
+			Optional:    true,
+		},
+		"config": {
+			Description: "The configuration for the connector. Configuration properties vary depending on the connector type.",
+			Type:        schema.TypeString,
+			Computed:    true,
+		},
+		"is_deprecated": {
+			Description: "Indicates whether the connector type is deprecated.",
+			Type:        schema.TypeBool,
+			Computed:    true,
+		},
+		"is_missing_secrets": {
+			Description: "Indicates whether secrets are missing for the connector.",
+			Type:        schema.TypeBool,
+			Computed:    true,
+		},
+		"is_preconfigured": {
+			Description: "Indicates whether it is a preconfigured connector.",
+			Type:        schema.TypeBool,
+			Computed:    true,
+		},
+	}
+
+	return &schema.Resource{
+		Description: "Search for a connector by name, space id, and type. Note, that this data source will fail if more than one connector shares the same name.",
+		ReadContext: datasourceConnectorRead,
+		Schema:      connectorSchema,
+	}
+}
+
+func datasourceConnectorRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, diags := clients.NewApiClientFromSDKResource(d, meta)
+	if diags.HasError() {
+		return diags
+	}
+	connectorName := d.Get("name").(string)
+	spaceId := d.Get("space_id").(string)
+	connectorType := d.Get("connector_type_id").(string)
+
+	foundConnectors, diags := kibana.SearchConnectors(ctx, client, connectorName, spaceId, connectorType)
+	if diags.HasError() {
+		return diags
+	}
+
+	if len(foundConnectors) == 0 {
+		diag.Errorf("error while creating elasticstack_kibana_action_connector datasource: connector with name [%s/%s] and type [%s] not found", spaceId, connectorName, connectorType)
+	}
+
+	if len(foundConnectors) > 1 {
```

Comment:
I do wonder if it would have been better for this DS to output a list of matching connectors TBH.

Reply:
For the use case that I want to use this with, I think this would cause confusion. What I'm trying to facilitate here is the ability to grab a connector by name (eg `o11y-alertbot`) and use it in an alert without having to create a new connector in the module. If the DS returns >1 connector then its ambiguous what connector is actually being used, particularly if it has been created erroneously.

Reply:
Yea, I understand. You could still achieve that by having a [check](https://developer.hashicorp.com/terraform/language/checks) block do the assertion on a single result type, whilst the DS would be more flexible for other consumers.